pkg/expect/expect.go

```diff
@@ -174,63 +213,89 @@ func (ep *ExpectProcess) LineCount() int {
 	return ep.count
 }
 
+// ExitCode returns the exit code of this process.
+// This will return the correct exit code after Close/Stop has been called, otherwise it returns math.MinInt
+func (ep *ExpectProcess) ExitCode() int {
+	ep.mu.Lock()
+	defer ep.mu.Unlock()
+
+	if ep.cmd == nil {
+		return ep.exitCode
+	}
+
+	return math.MinInt
+}
+
 // Stop kills the expect process and waits for it to exit.
 func (ep *ExpectProcess) Stop() error { return ep.close(true) }
 
 // Signal sends a signal to the expect process
 func (ep *ExpectProcess) Signal(sig os.Signal) error {
+	ep.mu.Lock()
+	defer ep.mu.Unlock()
+
+	if ep.cmd == nil {
+		return errors.New("expect process already closed")
+	}
+
 	return ep.cmd.Process.Signal(sig)
 }
 
 func (ep *ExpectProcess) Wait() error {
-	_, err := ep.cmd.Process.Wait()
-	return err
+	panic("who uses this")
```

Comment:
this is just temporary, a compile error regarding Wait() came out of nowhere, just interested what was trying to consume this. There's no interface around it...

Reply:
Turned out to be a change in main, I didn't know the action would apply this commit on top of main and run the tests on top of it. I've rebased and could fix it.